src/Bicep.Core/CoreResources.resx

```diff
@@ -223,55 +223,79 @@
     <value>Parameter "{0}" is declared but never used.</value>
     <comment>{0} is the paramenter name</comment>
-  </data>
+  </data>
 	<data name="PossibleSecretMessageFunction" xml:space="preserve">
     <value>function '{0}'</value>
     <comment>{0} is the function name. This string appears in the middle of a sentence, hence doesn't start with capital letter. Something similar to: Found: function 'listKeys()'</comment>
-  </data>
+  </data>
 	<data name="PossibleSecretMessageSecureParam" xml:space="preserve">
     <value>secure parameter '{0}'</value>
     <comment>{0} is the parameter name. This string appears in the middle of a sentence, so it doesn't start with a capital letter. Something similar to: Found: secure parameter 'parameter1'</comment>
-  </data>
+  </data>
 	<data name="SecureParameterDefaultFixTitle" xml:space="preserve">
     <value>Remove insecure default value.</value>
     <comment />
-  </data>
+  </data>
 	<data name="SecureParameterDefaultRuleDescription" xml:space="preserve">
     <value>Secure parameters should not have hardcoded defaults (except for empty or newGuid()).</value>
     <comment />
-  </data>
+  </data>
 	<data name="SimplifyInterpolationFixTitle" xml:space="preserve">
     <value>Remove unnecessary string interpolation.</value>
     <comment />
-  </data>
+  </data>
 	<data name="SimplifyInterpolationRuleDescription" xml:space="preserve">
     <value>Remove unnecessary string interpolation.</value>
     <comment />
-  </data>
+  </data>
 	<data name="UnusedVariableRuleDescription" xml:space="preserve">
     <value>All variables must be used.</value>
-  </data>
+  </data>
 	<data name="UnusedVariableRuleMessageFormat" xml:space="preserve">
     <value>Variable "{0}" is declared but never used.</value>
     <comment>{0} is the variable name</comment>
-  </data>
+  </data>
 	<data name="ProtectCommandToExecuteSecretsRuleDescription" xml:space="preserve">
     <value>Use protectedSettings for commandToExecute secrets</value>
-  </data>
+  </data>
 	<data name="ProtectCommandToExecuteSecretsRuleMessage" xml:space="preserve">
     <value>Use protectedSettings for commandToExecute secrets. Found possible secret: {0}</value>
     <comment>{0} is the description of what was found</comment>
-  </data>
+  </data>
 	<data name="UseStableVMImage" xml:space="preserve">
     <value>Virtual machines shouldn't use preview images.</value>
     <comment />
-  </data>
+  </data>
 	<data name="UseStableVMImageRuleFixMessageFormat" xml:space="preserve">
     <value>Virtual machines shouldn't use preview images. Use stable version in imageReference property "{0}".</value>
     <comment />
-  </data>
+  </data>
+  <data name="MaxNumberParametersRuleDescription" xml:space="preserve">
+    <value>Maximum number of parameters used.</value>
+  </data>
+  <data name="MaxNumberParametersRuleMessageFormat" xml:space="preserve">
+    <value>Too many parameters. Number of parameters is limited to 256.</value>

```

Comment:
>256 [](http://example.com/codeflow?start=67&length=3)

Could you update the strings so the limit is a parameter. We may increase the limits in the backend at some point and it'll make it easier for us to update if there's only one constant in the codebase for each limit.

Reply:
The same comment would apply to all the limits.